### ISSUE TITLE
Kwave is silent on playback

### ISSUE THREAD

**Issue by humdingerb** [haikuports/haikuports#5310]
I can load a WAV (or MP3) and see the wave form, but playing it back I hear nothing.
Under "Settings | Playback" the "Device" is empty.

Started from Terminal, I see upon playback:

```
--- meta data ---
* meta data #53-2020-10-06T20:22:54--{0e8bcc87-bdcb-4e6d-ac3e-2847d03a7d80}
    scope = signal
    'FILE_INFO: Bits per Sample' = '16'
    'FILE_INFO: Channels' = '2'
    'FILE_INFO: Compression' = '0'
    'FILE_INFO: File Size' = '3183722'
    'FILE_INFO: Filename' = '/HiQ-Data/Haiku contests/System sounds/Humdinger - System Sounds - Modulorian/Workfiles/Beep.wav'
    'FILE_INFO: Length' = '795919'
    'FILE_INFO: Mime Type' = 'audio/x-wav'
    'FILE_INFO: Sample Format' = '401'
    'FILE_INFO: Sample Rate' = '44100'
    'STDPROP_TYPE' = 'FILE INFO'
-----------------
playback method 'qt_audio' (0) not supported -> falling back to 'qt_audio' (2)
PlayBackPlugin::createDevice('qt_audio' [2])
PlayBackQt::open(device='', rate=44100.0,channels=2, bits=16, bufbase=13)
Kwave::PlayBackQt::close()
Kwave::PlayBackQt::close() - DONE
    buffer size = 8192
PlaybBackQt::stateChanged(ActiveState)
    period_size = 1764
    timeout = 100 ms
Kwave::PlayBackQt::Buffer::setTimeout(100)
Kwave::PlayBackQt::close()
```

Shaka on IRC states that he can choppily play back an MP3 on 64bit Haiku. 
Only silence for me on 32bit  hrev54606.

**Comment by humdingerb** [comment]
_Dario_ on IRC suggested to install "qtAV". Did that and entered under Settings Playback Device "/system/lib/x86/libQtAV.so.1. That resulted in an error mesage on playback, that the device is unknown or no longer exists.
After changing buffer size, playback "works", but is still silent.

**Comment by unspacyar** [comment]
Hello, _Dario_ here :-)

I guess that the recipe forget to declare the QtAV package as a dependence.  Anyways, after install it manually, I had to configurate it manually, like **humdingerb** said.   In my case, I get some sound playback, but it is extreme choppy. I tried playing with the buffer size, but I didn't get any improvement.

**Comment by threedeyes** [comment]
Hm. 
Kwave has native output and audio capture via MediaKit. I can't understand what has qt_audio and qtav got to do with it. I successfully record audio and play any audio files.
![2020-10-07_13-50](https://user-images.githubusercontent.com/2559260/95285465-32827000-08a4-11eb-968b-e5ff2b69d56a.png)
![2020-10-07_13-51](https://user-images.githubusercontent.com/2559260/95285466-33b39d00-08a4-11eb-8d52-fa52b4f02b1b.png)

**Comment by humdingerb** [comment]
I have a Beta2 64bit Haiku around here, and after installing Kwave (and its boatload of dependencies), it all worked without any settings fiddling. Though, contrary to your screenshots, I don't have anything listed under "Device".

The list of downloaded dependencies on 64bit is much larger than 32bit (though keep in mind, the 32bit nightly is my 'production' system, the 64bit Beta2 is barely touched):

```
32bit (nightly):

kwave_x86-20.08.0-1-x86_gcc2.hpkg
knotifications_x86-5.65.0-2-x86_gcc2.hpkg
phonon_gstreamer_x86-4.9.0-2-x86_gcc2.hpkg
phonon_x86-4.10.2-2-x86_gcc2.hpkg
libsamplerate_x86-0.1.9-2-x86_gcc2.hpkg
kitemmodels_x86-5.65.0-2-x86_gcc2.hpkg
audiofile_x86-0.3.6-6-x86_gcc2.hpkg
fftw_x86-3.3.8-5-x86_gcc2.hpkg
kactivities_x86-5.65.0-2-x86_gcc2.hpkg
kunitconversion_x86-5.65.0-2-x86_gcc2.hpkg
id3lib_x86-3.8.3-1-x86_gcc2.hpkg
kparts_x86-5.65.0-2-x86_gcc2.hpkg
libmad_x86-0.15.1b-7-x86_gcc2.hpkg
kdoctools_x86-5.65.0-2-x86_gcc2.hpkg


64bit (Beta2):

kwave-20.08.0-1-x86_64.hpkg
kparts-5.65.0-2-x86_64.hpkg
kio-5.65.0-2-x86_64.hpkg
kbookmarks-5.65.0-2-x86_64.hpkg
kxmlgui-5.65.0-2-x86_64.hpkg
ktextwidgets-5.65.0-2-x86_64.hpkg
kglobalaccel-5.65.0-2-x86_64.hpkg
kiconthemes-5.65.0-2-x86_64.hpkg
kservice-5.65.0-3-x86_64.hpkg
kconfigwidgets-5.65.0-2-x86_64.hpkg
knotifications-5.65.0-2-x86_64.hpkg
kcrash-5.65.0-2-x86_64.hpkg
kunitconversion-5.65.0-2-x86_64.hpkg
kcompletion-5.65.0-2-x86_64.hpkg
kjobwidgets-5.65.0-2-x86_64.hpkg
kitemmodels-5.65.0-2-x86_64.hpkg
kauth-5.65.0-2-x86_64.hpkg
kactivities-5.65.0-2-x86_64.hpkg
kdoctools-5.65.0-2-x86_64.hpkg
phonon_gstreamer-4.9.0-2-x86_64.hpkg
kwindowsystem-5.65.0-2-x86_64.hpkg
solid-5.65.0-2-x86_64.hpkg
ki18n-5.65.0-2-x86_64.hpkg
kcodecs-5.65.0-2-x86_64.hpkg
kguiaddons-5.65.0-2-x86_64.hpkg
kconfig-5.65.0-2-x86_64.hpkg
kwidgetsaddons-5.65.0-2-x86_64.hpkg
kcoreaddons-5.65.0-2-x86_64.hpkg
sonnet-5.65.0-2-x86_64.hpkg
karchive-5.65.0-2-x86_64.hpkg
kitemviews-5.65.0-2-x86_64.hpkg
kdbusaddons-5.65.0-2-x86_64.hpkg
attica-5.65.0-2-x86_64.hpkg
phonon-4.10.2-2-x86_64.hpkg
qthaikuplugins-5.15.1.2-1-x86_64.hpkg
qt5-5.15.1-2-x86_64.hpkg
gst_plugins_bad-1.18.0-1-x86_64.hpkg
gst_plugins_ugly-1.18.0-1-x86_64.hpkg
gst_libav-1.18.0-1-x86_64.hpkg
gst_plugins_base-1.18.0-1-x86_64.hpkg
fluidsynth2-2.1.1-2-x86_64.hpkg
libnice-0.1.16-1-x86_64.hpkg
gstreamer-1.18.0-1-x86_64.hpkg
libmms-0.6.4-3-x86_64.hpkg
libinstpatch-1.1.3-2-x86_64.hpkg
twolame-0.3.13-4-x86_64.hpkg
mpg123-1.25.13-1-x86_64.hpkg
gnutls-3.6.10-2-x86_64.hpkg
glib2-2.62.0-3-x86_64.hpkg
aspell_en-2018.04.16~0-1-any.hpkg
libsndfile-1.0.28-1-x86_64.hpkg
libtool_libltdl-2.4.6-2-x86_64.hpkg
a52dec-0.7.4-5-x86_64.hpkg
libidn2-2.0.5-1-x86_64.hpkg
libunistring-0.9.10-1-x86_64.hpkg
p11_kit-0.23.18.1-1-x86_64.hpkg
libsdl-1.2.15-17-x86_64.hpkg
tremor-1.0.0~git-1-x86_64.hpkg
gettext-0.19.8.1-7-x86_64.hpkg
rtmpdump-2.4_20161210-6-x86_64.hpkg
nettle-3.5.1-1-x86_64.hpkg
faad2-2.9.1-2-x86_64.hpkg
orc-0.4.30-1-x86_64.hpkg
x265-3.2-1-x86_64.hpkg
libsrtp-1.5.4-2-x86_64.hpkg
qsystray-5.14.2.0-1-x86_64.hpkg
double_conversion-3.1.5-1-x86_64.hpkg
flite-2.1-1-x86_64.hpkg
icu66-66.1-1-x86_64.hpkg
openal-1.19.1-1-x86_64.hpkg
aspell-0.60.8-1-x86_64.hpkg
libmad-0.15.1b-7-x86_64.hpkg
flac-1.3.3-1-x86_64.hpkg
libsamplerate-0.1.9-2-x86_64.hpkg
fftw-3.3.8-5-x86_64.hpkg
audiofile-0.3.6-6-x86_64.hpkg
opus-1.3.1-1-x86_64.hpkg
id3lib-3.8.3-1-x86_64.hpkg
```

**Comment by extrowerk** [comment]
On 64 bit recent Haiku: MediaKit Input available as recording device, but the playback device list is empty.

**Comment by threedeyes** [comment]
I reproduce this bug.
I think kwave updates the list of QtMultimedia's audio devices when the window is closed by clicking OK. If you open the window for the first time - MediaKit will be missing. But if you click OK and enter the settings dialog again, it will appear and the sound starts working.

**Comment by unspacyar** [comment]
@threedeyes: thanks for the hint, and sorry if I made some asumption that leaded to an incorrect diagnostic.  I installed Kwave in another machine (also Haiku 32bits), and after click "OK" in the empty configuration window and reopened,  now I get the MediaKit device.

However, in this machine I had the same problem Humdingerb said: there is no sound output.
In the other machine, the sound appears but only after leaving it playing for about 15 seconds.  For example, if I open an mp3 file and put it to play, for the first 15 seconds (more or less), I didn't get any output. But after that, the sound apperars.

**Comment by unspacyar** [comment]
There is the Terminal output, from the machine were there is no sound:

SignalManager::loadFile(file:///boot/home/Wav_868kb.wav) - [audio/x-wav]
QImage: XPM pixels missing on image line 0 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 1 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 2 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 3 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 4 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 5 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 6 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 7 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 8 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 9 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 10 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 11 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 12 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 13 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 14 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 15 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 0 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 1 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 2 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 3 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 4 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 5 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 6 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 7 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 8 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 9 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 10 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 11 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 12 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 13 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 14 (possibly a C++ trigraph).
QImage: XPM pixels missing on image line 15 (possibly a C++ trigraph).
--- meta data ---
* meta data #53-2020-10-07T21:22:19--{b95d3e60-d02f-4bf0-a998-2788b1324693}
    scope = signal
    'FILE_INFO: Bits per Sample' = '16'
    'FILE_INFO: Channels' = '2'
    'FILE_INFO: Compression' = '0'
    'FILE_INFO: File Size' = '889422'
    'FILE_INFO: Filename' = '/boot/home/Wav_868kb.wav'
    'FILE_INFO: Length' = '222336'
    'FILE_INFO: Mime Type' = 'audio/x-wav'
    'FILE_INFO: Sample Format' = '401'
    'FILE_INFO: Sample Rate' = '44100'
    'FILE_INFO: Software' = 'Lavf58.35.100'
    'STDPROP_TYPE' = 'FILE INFO'
-----------------
// here I push the "Play" button on KWave:

PlayBackPlugin::createDevice('qt_audio' [2])
PlayBackQt::open(device='MediaKit Output|sound_note', rate=44100.0,channels=2, bits=16, bufbase=18)
Kwave::PlayBackQt::close()
Kwave::PlayBackQt::close() - DONE
    buffer size = 262144
PlaybBackQt::stateChanged(ActiveState)
    period_size = 1764
    timeout = 2972 ms
Kwave::PlayBackQt::Buffer::setTimeout(2972)
Kwave::PlayBackQt::close()
PlaybBackQt::stateChanged(StoppedState)
Kwave::PlayBackQt::Buffer::readData(...) -> read=6372/16384
Kwave::PlayBackQt::close() - flushing..., state=2
Kwave::PlayBackQt::close() - flushing done.
Kwave::PlayBackQt::close() - DONE

**Comment by threedeyes** [comment]
hm, looks good - mediakit output detected: (PlayBackQt::open(device='MediaKit Output|sound_note')
Are there any playback problems in other programs? qmplay2 for example

**Comment by threedeyes** [comment]
Try to play with buffer size parameters in KWave playback settings.

**Comment by humdingerb** [comment]
Aha! Kwave does play back with sound on 32bit, too, just as @unspacyar said, after a long while. I didn't see that before, because my clips where only 1s long...

**Comment by unspacyar** [comment]
Hi!  I tested in two machines (both 32 bit Haiku). In one of it, I have the exact behaviour that @humdingerb said (the sound start after several seconds after push the "play" button.  But in the second machine, I didn't managed to get any sound. I tried by changing the buffer size, but without effect.

The last test will be to install 64 bit Haiku in the same machine.  I will update this thread with the results.  Thank you!
